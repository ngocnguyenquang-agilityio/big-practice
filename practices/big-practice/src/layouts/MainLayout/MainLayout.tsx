interface IMainLayout {
  children: React.ReactNode;
  header: React.ReactNode;
  footer: React.ReactNode;
}

const MainLayout = ({ children, header, footer }: IMainLayout) => {
  return (
    <div className='bg-neutral-900 text-white selection:bg-pink-500 selection:text-white'>
      {header}
      <div className='mx-auto flex max-w-screen-2xl flex-col gap-8 px-4 pb-4 text-white md:flex-row'>{children}</div>
      {footer}
    </div>
  );
};

export default MainLayout;
