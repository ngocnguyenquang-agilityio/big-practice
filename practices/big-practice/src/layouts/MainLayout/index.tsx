// Components
import Cart from '@components/Cart';
import Footer from '@components/Footer';
import Header from '@components/Header';

// Stores
import { useCart } from '@stores/cart';

const MainLayout = ({ children }: { children: React.ReactNode }) => {
  const { isOpenCart, toggleCart } = useCart();

  return (
    <div className='bg-neutral-900 text-white selection:bg-pink-500 selection:text-white'>
      <Header onToggleCart={toggleCart} />
      <div className='mx-auto flex max-w-screen-2xl flex-col gap-8 px-4 pb-4 text-white md:flex-row'>{children}</div>
      <Footer />
      {isOpenCart && <Cart handleToggleCart={toggleCart} />}
    </div>
  );
};

export default MainLayout;
