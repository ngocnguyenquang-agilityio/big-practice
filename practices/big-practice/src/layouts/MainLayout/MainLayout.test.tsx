import { MemoryRouter } from 'react-router-dom';
import { RenderResult, render } from '@testing-library/react';

import MainLayout from '.';

const MockHeader = () => <div>Header</div>;
const MockFooter = () => <div>Footer</div>;

const propsMainLayout = {
  header: <MockHeader />,
  footer: <MockFooter />,
};

describe('MainLayout', () => {
  let comp: RenderResult;

  beforeEach(() => {
    comp = render(
      <MemoryRouter>
        <MainLayout {...propsMainLayout}>
          <div>Children</div>
        </MainLayout>
      </MemoryRouter>,
    );
  });

  test('Should render correctly', () => {
    expect(comp).toBeTruthy();
  });

  test('Should render correctly', () => {
    expect(comp).toMatchSnapshot();
  });
});
