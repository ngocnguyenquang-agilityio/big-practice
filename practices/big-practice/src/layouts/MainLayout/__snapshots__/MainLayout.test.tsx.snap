// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-neutral-900 text-white selection:bg-pink-500 selection:text-white"
      >
        <div
          class="relative flex items-center justify-between p-4 lg:px-6"
        >
          <div
            class="flex w-full items-center"
          >
            <div
              class="flex w-full md:w-1/3 items-center"
            >
              <div
                class="mr-2 flex w-full items-center justify-center md:w-auto lg:mr-6"
              >
                <button
                  class="inline-flex items-center text-white justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 h-11 w-11 rounded-md"
                >
                  <div
                    class="flex flex-none items-center justify-center border border-neutral-700 bg-black h-[40px] w-[40px] rounded-xl"
                  >
                    <svg
                      aria-label="Acme Store logo"
                      class="h-5 w-5 fill-white h-[16px] w-[16px]"
                      name="logo-icon"
                      viewBox="0 0 32 28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M21.5758 9.75769L16 0L0 28H11.6255L21.5758 9.75769Z"
                      />
                      <path
                        d="M26.2381 17.9167L20.7382 28H32L26.2381 17.9167Z"
                      />
                    </svg>
                  </div>
                </button>
                <a
                  class="ml-2 text-sm font-medium uppercase text-white"
                  href=""
                >
                  acme store
                </a>
              </div>
              <ul
                class="gap-6 text-sm md:flex md:items-center"
              >
                <li
                  class="underline-offset-4 hover:underline text-neutral-400 hover:text-neutral-300 cursor-pointer"
                >
                  All
                </li>
                <li
                  class="underline-offset-4 hover:underline text-neutral-400 hover:text-neutral-300 cursor-pointer"
                >
                  Shirts
                </li>
              </ul>
            </div>
            <div
              class="justify-center md:flex md:w-1/3"
            >
              <div
                class="relative w-full"
              >
                <input
                  class="w-full rounded-lg border border-neutral-800 bg-transparent px-4 py-2 text-sm text-white placeholder:text-neutral-400"
                  placeholder="Search Products..."
                  type="text"
                />
                <div
                  class="absolute inset-y-0 right-0 flex items-center pr-4 pointer-events-none"
                >
                  <svg
                    aria-hidden="true"
                    class="w-4 h-4 text-white"
                    fill="none"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="flex justify-end md:w-1/3"
            >
              <button
                class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 border border-neutral-700 transition-colors h-11 w-11 rounded-md"
              >
                <svg
                  class="lucide lucide-shopping-cart"
                  fill="none"
                  height="16"
                  name="cart-icon"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="8"
                    cy="21"
                    r="1"
                  />
                  <circle
                    cx="19"
                    cy="21"
                    r="1"
                  />
                  <path
                    d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="mx-auto flex max-w-screen-2xl flex-col gap-8 px-4 pb-4 text-white md:flex-row"
        />
        <footer
          class="text-sm text-neutral-400"
        >
          <div
            class="mx-auto flex w-full max-w-7xl flex-col gap-6 border-t px-6 py-12 text-sm border-neutral-700 md:flex-row md:gap-12 md:px-4 min-[1320px]:px-0"
          >
            <div>
              <button
                class="inline-flex items-center text-white justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 h-11 w-11 rounded-md"
              >
                <div
                  class="flex flex-none items-center justify-center border border-neutral-700 bg-black h-[40px] w-[40px] rounded-xl"
                >
                  <svg
                    aria-label="Acme Store logo"
                    class="h-5 w-5 fill-white h-[16px] w-[16px]"
                    name="logo-icon"
                    viewBox="0 0 32 28"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21.5758 9.75769L16 0L0 28H11.6255L21.5758 9.75769Z"
                    />
                    <path
                      d="M26.2381 17.9167L20.7382 28H32L26.2381 17.9167Z"
                    />
                  </svg>
                </div>
              </button>
            </div>
            <div>
              <nav>
                <ul>
                  <li
                    class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                  >
                    Home
                  </li>
                  <li
                    class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                  >
                    About
                  </li>
                  <li
                    class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                  >
                    Terms & Conditions
                  </li>
                  <li
                    class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                  >
                    Privacy Policy
                  </li>
                  <li
                    class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                  >
                    FAQ
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="md:ml-auto"
            >
              <a
                class="flex h-8 w-max flex-none items-center justify-center rounded-md border text-xs border-neutral-700 bg-black text-white"
              >
                <span
                  class="px-3"
                >
                  â–²
                </span>
                <hr
                  class="h-full border-r border-neutral-700"
                />
                <span
                  class="px-3"
                >
                  Deploy
                </span>
              </a>
            </div>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-neutral-900 text-white selection:bg-pink-500 selection:text-white"
    >
      <div
        class="relative flex items-center justify-between p-4 lg:px-6"
      >
        <div
          class="flex w-full items-center"
        >
          <div
            class="flex w-full md:w-1/3 items-center"
          >
            <div
              class="mr-2 flex w-full items-center justify-center md:w-auto lg:mr-6"
            >
              <button
                class="inline-flex items-center text-white justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 h-11 w-11 rounded-md"
              >
                <div
                  class="flex flex-none items-center justify-center border border-neutral-700 bg-black h-[40px] w-[40px] rounded-xl"
                >
                  <svg
                    aria-label="Acme Store logo"
                    class="h-5 w-5 fill-white h-[16px] w-[16px]"
                    name="logo-icon"
                    viewBox="0 0 32 28"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21.5758 9.75769L16 0L0 28H11.6255L21.5758 9.75769Z"
                    />
                    <path
                      d="M26.2381 17.9167L20.7382 28H32L26.2381 17.9167Z"
                    />
                  </svg>
                </div>
              </button>
              <a
                class="ml-2 text-sm font-medium uppercase text-white"
                href=""
              >
                acme store
              </a>
            </div>
            <ul
              class="gap-6 text-sm md:flex md:items-center"
            >
              <li
                class="underline-offset-4 hover:underline text-neutral-400 hover:text-neutral-300 cursor-pointer"
              >
                All
              </li>
              <li
                class="underline-offset-4 hover:underline text-neutral-400 hover:text-neutral-300 cursor-pointer"
              >
                Shirts
              </li>
            </ul>
          </div>
          <div
            class="justify-center md:flex md:w-1/3"
          >
            <div
              class="relative w-full"
            >
              <input
                class="w-full rounded-lg border border-neutral-800 bg-transparent px-4 py-2 text-sm text-white placeholder:text-neutral-400"
                placeholder="Search Products..."
                type="text"
              />
              <div
                class="absolute inset-y-0 right-0 flex items-center pr-4 pointer-events-none"
              >
                <svg
                  aria-hidden="true"
                  class="w-4 h-4 text-white"
                  fill="none"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="flex justify-end md:w-1/3"
          >
            <button
              class="inline-flex items-center text-white justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 border border-neutral-700 transition-colors h-11 w-11 rounded-md"
            >
              <svg
                class="lucide lucide-shopping-cart"
                fill="none"
                height="16"
                name="cart-icon"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="21"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="21"
                  r="1"
                />
                <path
                  d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="mx-auto flex max-w-screen-2xl flex-col gap-8 px-4 pb-4 text-white md:flex-row"
      />
      <footer
        class="text-sm text-neutral-400"
      >
        <div
          class="mx-auto flex w-full max-w-7xl flex-col gap-6 border-t px-6 py-12 text-sm border-neutral-700 md:flex-row md:gap-12 md:px-4 min-[1320px]:px-0"
        >
          <div>
            <button
              class="inline-flex items-center text-white justify-center text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-slate-950 focus-visible:ring-slate-300 underline-offset-4 hover:underline dark:text-neutral-100 h-11 w-11 rounded-md"
            >
              <div
                class="flex flex-none items-center justify-center border border-neutral-700 bg-black h-[40px] w-[40px] rounded-xl"
              >
                <svg
                  aria-label="Acme Store logo"
                  class="h-5 w-5 fill-white h-[16px] w-[16px]"
                  name="logo-icon"
                  viewBox="0 0 32 28"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.5758 9.75769L16 0L0 28H11.6255L21.5758 9.75769Z"
                  />
                  <path
                    d="M26.2381 17.9167L20.7382 28H32L26.2381 17.9167Z"
                  />
                </svg>
              </div>
            </button>
          </div>
          <div>
            <nav>
              <ul>
                <li
                  class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                >
                  Home
                </li>
                <li
                  class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                >
                  About
                </li>
                <li
                  class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                >
                  Terms & Conditions
                </li>
                <li
                  class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                >
                  Privacy Policy
                </li>
                <li
                  class="block p-2 text-lg underline-offset-4 hover:underline hover:text-neutral-300 md:inline-block md:text-sm cursor-pointer"
                >
                  FAQ
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="md:ml-auto"
          >
            <a
              class="flex h-8 w-max flex-none items-center justify-center rounded-md border text-xs border-neutral-700 bg-black text-white"
            >
              <span
                class="px-3"
              >
                â–²
              </span>
              <hr
                class="h-full border-r border-neutral-700"
              />
              <span
                class="px-3"
              >
                Deploy
              </span>
            </a>
          </div>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
